src/Code.gs

```diff
@@ -553,36 +580,38 @@ FirebaseApp_._sendAllRequests = function (finalRequests, originalsRequests, db,
     
     var responseContent = responses[i].getContentText();
     
-    // Avoid returning the Firebase app secret in case of error
+    // if response content is a string and contains the Firebase secret, assume it's an error on which a retry is needed
+    // and replace the error returned by a generic one to avoid throwing the secret
     if (db.base.secret && typeof responseContent === 'string' && responseContent.indexOf(db.base.secret) !== -1){
       errorCount += 1;
       
-      originalsRequests[i].error = new Error(FirebaseApp_._ERROR_TRY_AGAIN);
+      originalsRequests[i].error = new Error(FirebaseApp_.NORMALIZED_ERRORS.TRY_AGAIN);
       
       retry.finalReq.push(finalRequests[i]);
       retry.originalReq.push(originalsRequests[i]);
       
       continue;
     }
     
-    
-    // try parsing response
     var errorMessage;
     var responseParsed;
+    // try parsing response
     try{
       responseParsed = JSON.parse(responseContent);
     }
     catch(e){
-      errorMessage = FirebaseApp_._ERROR_TRY_AGAIN;
-      // if responseContent is undefined => internal error sur UrlFetch service, try again
-      // It will be caught as JSON.parse(undefined) will fail
+      // if responseContent is undefined => internal error on UrlFetch service, try again
+      // It is caught as JSON.parse(undefined) fails ("Unexpected token")
+      errorMessage = FirebaseApp_.NORMALIZED_ERRORS.TRY_AGAIN;
     }
     
-    // Process possible errors and retry
-    if (FirebaseApp_._errorCodeList[responseCode] || errorMessage) {
+    // Retry on specific response codes, specific error messages or if we failed to parse the response
+    if (FirebaseApp_._errorCodeList[responseCode] || !(responseParsed && responseParsed.error && FirebaseApp_.NORETRY_ERRORS[responseParsed.error]) || errorMessage) {
       errorCount += 1;
       
-      originalsRequests[i].error = new Error(errorMessage || (responseParsed && responseParsed.error) || FirebaseApp_._ERROR_TRY_AGAIN);
+      // Add the response code to the error message if it comes from the response
+      if (responseParsed && responseParsed.error) originalsRequests[i].error = new Error(responseCode + " - " + responseParsed.error);
```

Comment:
Not sure if it's the best way to handle that but if would be useful to get the response code as part of the error.